app/src/androidTest/java/com/github/ravifrancesco/softwaredevproject/UserPointTest.java

```diff
@@ -0,0 +1,87 @@
+package com.github.ravifrancesco.softwaredevproject;
+
+import android.content.Context;
+import android.content.Intent;
+
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.rule.ServiceTestRule;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.Random;
+import java.util.concurrent.TimeoutException;
+
+
+public class UserPointTest {
+
+    @Rule
+    public final ServiceTestRule serviceRule = new ServiceTestRule();
+
+    @Test
+    public void getLocationTest() throws TimeoutException {
```

Comment:
If I understand the test correclty it tests that the default value is working not the getLocation funciton. If that is the case then maybe change testname. 
```suggestion
    public void defaultLocationTest() throws TimeoutException {
```

Reply:
It checks if the GPStracker is able to get the location (if it is, I cannot check if the location is correct, so the test just passes). If it is not able to get the location the tests check if the default location is returned